lib/sms_template_finder.rb

```diff
@@ -0,0 +1,33 @@
+class SmsTemplateFinder
+  def execute(message_content:, env:)
+    device_name_matchers.each do |matcher, device_name|
+      return device_instruction_config(env).fetch(device_name) if message_content.match?(matcher)
+    end
+
+    template_name = 'generic_help'
+    template_name = 'credentials' if message_content == '' || message_content.match(/go/i)
+    template_name = 'help_menu' if message_content == 'help'
+
+    config(env)[template_name]
+  end
+
+private
+
+  def device_instruction_config(env)
+    config(env).fetch('device_help')
+  end
+
+  def device_name_matchers
+    {
+      /1|android|samsung|galaxy|htc|huawei|sony|motorola|lg|nexus/i => 'android',
```

Comment:
Would be nice to see all these regexes grouped and assigned to constants with descriptive names.